CMakeLists.txt

```diff
@@ -22,6 +22,42 @@ string(REGEX REPLACE ".*#define ONEDPL_VERSION_MAJOR ([0-9]+).*" "\\1" _onedpl_v
 string(REGEX REPLACE ".*#define ONEDPL_VERSION_MINOR ([0-9]+).*" "\\1" _onedpl_ver_minor "${_onedpl_version_info}")
 string(REGEX REPLACE ".*#define ONEDPL_VERSION_PATCH ([0-9]+).*" "\\1" _onedpl_ver_patch "${_onedpl_version_info}")
 
+# CMAKE_CXX_COMPILER and CMAKE_CXX_COMPILER_VERSION cannot be used because
+# CMake 3.19 and older will detect IntelLLVM compiler as CLang with CLang-specific version, see https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
+if (CMAKE_CXX_COMPILER MATCHES ".*(dpcpp-cl|dpcpp|icx-cl|icpx)(.exe)?$")
+    set(INTEL_LLVM_COMPILER TRUE)
+    execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_VARIABLE INTEL_LLVM_COMPILER_VERSION_RAW)
+    string(REGEX MATCH "[0-9][0-9][0-9][0-9]\\.[0-9]\\.[0-9]" INTEL_LLVM_COMPILER_VERSION ${INTEL_LLVM_COMPILER_VERSION_RAW})
+    if (CMAKE_CXX_COMPILER MATCHES ".*(dpcpp-cl|icx-cl)(.exe)?$")
+        set(INTEL_LLVM_COMPILER_MSVC_LIKE TRUE)
+        set(MSVC TRUE)
+    else()
+        set(INTEL_LLVM_COMPILER_GNU_LIKE TRUE)
+    endif()
+endif()
+
+#Always turn off optimization in debug mode
+set(CMAKE_CXX_FLAGS_DEBUG_INIT "-O0")
```

Comment:
Let's move that inside `if (WIN32 AND INTEL_LLVM_COMPILER_GNU_LIKE)` condition and rely on the default cmake behavior for a certain compiler. For example, I believe cmake sets `/Od` for Microsof Visual C++ compiler.

Reply:
Yes, you are correct.  I will re-add the similar line to the else below for `if (MSVC)`.